ibis/backends/materialize/__init__.py

```diff
@@ -0,0 +1,2832 @@
+        -------
+        list[str]
+            List of connection names
+
+        Examples
+        --------
+        >>> import ibis
+        >>> con = ibis.materialize.connect()
+        >>> con.list_connections()
+        ['kafka_conn', 'pg_conn', 'aws_conn']
+        >>> con.list_connections(like="kafka%")
+        ['kafka_conn']
+        """
+        query = """
+        SELECT c.name
+        FROM mz_catalog.mz_connections c
+        JOIN mz_catalog.mz_schemas s ON c.schema_id = s.id
+        WHERE s.name = %(database)s
+        """
+
+        params = {"database": database or self.current_database}
+
+        if like is not None:
+            query += " AND c.name LIKE %(like)s"
+            params["like"] = like
+
+        query += " ORDER BY c.name"
+
+        with self.begin() as cur:
+            cur.execute(query, params)
+            return [row[0] for row in cur.fetchall()]
+
+    def alter_connection(
+        self,
+        name: str,
+        /,
+        *,
+        set_options: dict[str, str | Any] | None = None,
+        reset_options: list[str] | None = None,
+        rotate_keys: bool = False,
+        database: str | None = None,
+        schema: str | None = None,
+    ) -> None:
+        """Alter a connection's configuration.
+
+        Modify connection parameters, reset them to defaults, or rotate SSH tunnel
+        keys.
+
+        Parameters
+        ----------
+        name
+            Connection name to alter.
+        set_options
+            Dictionary of connection options to set. Options depend on connection
+            type:
+            - Kafka: 'BROKER', 'SASL MECHANISMS', etc.
+            - Postgres: 'HOST', 'PORT', 'DATABASE', etc.
+            - AWS: 'REGION', 'ACCESS KEY ID', etc.
+            Values can be strings or SECRET() references.
+        reset_options
+            List of option names to reset to defaults.
+        rotate_keys
+            If True, rotate SSH tunnel key pairs. Only valid for SSH TUNNEL
+            connections. Requires manual update of SSH bastion server keys.
+        database
+            Name of the database (catalog) where the connection exists.
+        schema
+            Name of the schema where the connection exists.
+
+        Examples
+        --------
+        >>> import ibis
+        >>> con = ibis.materialize.connect()
+
+        Update connection broker:
+
+        >>> con.alter_connection("kafka_conn", set_options={"BROKER": "new-broker:9092"})
+
+        Reset connection port to default:
+
+        >>> con.alter_connection("pg_conn", reset_options=["PORT"])
+
+        Rotate SSH tunnel keys:
+
+        >>> con.alter_connection("ssh_conn", rotate_keys=True)
+
+        Notes
+        -----
+        - Changes are applied atomically
+        - Cannot modify the same parameter via both SET and RESET
+        - For SSH key rotation, update bastion server keys manually after rotation
+
+        """
+        if not any([set_options, reset_options, rotate_keys]):
```

Comment:
```suggestion
        if not (set_options or reset_options or rotate_keys):
```

might actually be shorter and it's not constructing a list.